### PR TITLE
Make zk connection retries configurable

### PR DESCRIPTION
**Pull request by a2l007**
### Description

This makes the zookeeper connection retry count configurable. This is presently hardcoded to 29 tries which ends up taking a long time for the druid node to shutdown in case of ZK connectivity loss. 
Having a shorter retry count helps k8s deployments to fail fast. In situations where the underlying k8s node loses network connectivity or is no longer able to talk to zookeeper, failing fast can trigger pod restarts which can then reassign the pod to a healthy k8s node.  
Existing behavior is preserved, but users can override this property if needed.

<hr>

##### Key changed/added classes in this PR
 * `CuratorConfig`

This PR has:

- [x] been self-reviewed.
- [x] added documentation for new or modified features or behaviors.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

**Review by cryptoe — APPROVED** [review]
LGTM!!

**Comment by a2l007** [comment]
@abhishekagarwal87 We have this currently set to 5 on clusters which prompts a pod restart in ~30 seconds. Maybe we could reduce the default retry count and also keep it configurable for a release or so and then remove the config after that? That gives some time to see if a shorter retry count is leading to any negative impact for other users.

**Comment by abhishekagarwal87** [comment]
ok. in that case, lets not document the config in docs. You can just add a javadoc in the config class. does that sound good?

**Comment by a2l007** [comment]
@abhishekagarwal87 Sounds good to me. I've made the needed changes

File: server/src/main/java/org/apache/druid/curator/CuratorConfig.java [hunk 1]
```diff
@@ -58,6 +58,12 @@
   @JsonProperty("authScheme")
   private String authScheme = "digest";
 
+  // Configures the maximum number of retries for attempting connection to Zookeeper.
+  // This config property is in place to override the retry count in case unexpected issues show up with a smaller retry count.
+  // This may be removed in the future if we determine the default value doesn't cause any unexpected issues.
+  @JsonProperty("maxZkRetries")
+  private int maxZkRetries = 5;
```
Comment (abhishekagarwal87) [inline server/src/main/java/org/apache/druid/curator/CuratorConfig.java:65]: let's keep the default to 29 itself.

Comment (a2l007) [reply server/src/main/java/org/apache/druid/curator/CuratorConfig.java]: done

**Review by abhishekagarwal87 — APPROVED** [review]

**Pull request merged**